source/driverHandler.py

```diff
@@ -0,0 +1,204 @@
+
+	L{supportedSettings} should be set as appropriate for the settings supported by the driver.
+	Each setting is retrieved and set using attributes named after the setting;
+	e.g. the L{dotFirmness} attribute is used for the L{dotFirmness} setting.
+	These will usually be properties.
+	@ivar supportedSettings: The settings supported by the driver.
+	@type supportedSettings: list or tuple of L{DriverSetting}
+	"""
+
+	#: The name of the driver; must be the original module file name.
+	#: @type: str
+	name = ""
+	#: A description of the driver.
+	#: @type: str
+	description = ""
+	#: The configuration section where driver specific subsections should be saved.
+	#: @type: str
+	configSection = ""
+
+	def __init__(self):
+		"""Initialize this driver.
+		This method can also set default settings for the driver.
+		@raise Exception: If an error occurs.
+		@postcondition: This driver can be used.
+		"""
+
+	def terminate(self):
+		"""Terminate this driver.
+		This should be used for any required clean up.
+		@precondition: L{initialize} has been called.
+		@postcondition: This driver can no longer be used.
+		"""
+
+	def _get_supportedSettings(self):
+		raise NotImplementedError
+
+	@classmethod
+	def check(cls):
+		"""Determine whether this driver is available.
+		The driver will be excluded from the list of available drivers if this method returns C{False}.
+		For example, if a speech synthesizer requires installation and it is not installed, C{False} should be returned.
+		@return: C{True} if this driver is available, C{False} if not.
+		@rtype: bool
+		"""
+		return False
+
+	def isSupported(self,settingName):
+		"""Checks whether given setting is supported by the driver.
+		@rtype: l{bool}
+		"""
+		for s in self.supportedSettings:
+			if s.name==settingName: return True
+		return False
+
+	def getConfigSpec(self):
+		spec=deepcopy(config.confspec[self.configSection]["__many__"])
+		for setting in self.supportedSettings:
+			spec[setting.name]=setting.configSpec
+		return spec
+
+	def saveSettings(self):
+		conf=config.conf[self.configSection][self.name]
+		# Make sure the config spec is up to date, so the config validator does its work.
+		conf.spec.update(self.getConfigSpec())
+		for setting in self.supportedSettings:
+			try:
+				conf[setting.name]=getattr(self,setting.name)
+			except UnsupportedConfigParameterError:
+				log.debugWarning("Unsupported setting %s; ignoring"%s.name, exc_info=True)
+				continue
+
+	def loadSettings(self, onlyChanged=False):
+		c=config.conf[self.configSection][self.name]
+		for s in self.supportedSettings:
+			if c.get(s.name) is None:
+				continue
+			val=c[s.name]
+			if onlyChanged and getattr(self,s.name)==val:
+				continue
+			try:
+				setattr(self,s.name,val)
+			except UnsupportedConfigParameterError:
+				log.debugWarning("Unsupported setting %s; ignoring"%s.name, exc_info=True)
+				continue
+
+	@classmethod
+	def _paramToPercent(cls, current, min, max):

```

Comment:
The problem is that they are used within many synthesizer classes, moving them out of the SynthDriver or one of its parent classes would cause backwards incompatibility. Having said that, I can move them back to the SynthDriver class if you desire.

Reply:
Why can't you do the same as you already have done, import it for backwards compatibility?